transpiler/include/nil/blueprint/transpiler/aggregated_verifier_generator.hpp

```diff
@@ -0,0 +1,296 @@
+                    bool map_after_first = false;
+                    for (const auto &initial_proofs_map : initial_proofs) {
+                        if (map_after_first) [[likely]] out << "," << std::endl;
+                        out << "\t\t\t{\"struct\":[" << std::endl;
+                        bool inner_after_first = false;
+                        for (const auto &[index, value] : initial_proofs_map) {
+                            if (inner_after_first) [[likely]] out << "," << std::endl;
+                            // each initial proof is polynomials_values_type values;
+                            // which is std::vector<std::vector<std::array<value_type, FRI::m>>>
+                            // and merkle_proof_type p;
+                            // first the values
+                            // I have no idea if we have to store keys, I hope we do not
+                            const auto &values = value.values;
+                            bool outer_vec_after_first = false;
+                            for (const auto &outer_vector : values) {
+                                if (outer_vec_after_first) [[likely]] out << "," << std::endl;
+                                out << "\t\t\t\t{\"array\":[" << std::endl;
+                                bool core_after_first = false;
+                                for (const auto &inner_vector : outer_vector) {
+                                    if (core_after_first) [[likely]] out << "," << std::endl;
+                                    out << "\t\t\t\t\t{\"array\":[" << std::endl;
+                                    bool array_after_first = false;
+                                    for (const auto &elem : inner_vector) {
+                                        if (array_after_first) [[likely]] out << "," << std::endl;
+                                        out << "\t\t\t\t\t\t{\"field\":\"" << elem << "\"}";
+                                        array_after_first = true;
+                                    }
+                                    out << std::endl << "\t\t\t\t\t]}";
+                                    core_after_first = true;
+                                }
+                                out << std::endl << "\t\t\t\t]}";
+                                outer_vec_after_first = true;
+                            }
+                            // we skip the merkle proof, as those are the same for all initial proofs
+                            // we serialize it right after this loop
+                            inner_after_first = true;
+                        }
+                        out << "\t\t\t]}" << std::endl;
+                        map_after_first = true;
+                    }
+                    out << "\t\t]}" << std::endl;
+                    after_first = true;
+                }
+                out << "\t\t]}," << std::endl;
+                // and now serialize one of the merkle proofs
```

Comment:
Merkle proof consists of pairs <hash, position>, where position is {0,1}
We need hashes for all initial proofs.

Positions should be stored only once.

Reply:
Should be fixed now.